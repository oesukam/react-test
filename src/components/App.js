import React, { Component } from "react";
import { Button } from "react-bootstrap";
import uuid from "uuid";

class App extends Component {
  constructor() {
    super();
    this.state = {
      gifts: []
    };
  }
  addGift = () => {
    const { gifts } = this.state;
    gifts.push({ id: uuid.v4() });
    this.setState({
      gifts
    });
  };
  renderGifts = () => {
    return this.state.gifts.map(gift => <div key={gift.id}>gift</div>);
  };
  render() {
    return (
      <div>
        <h2>Gift Giver</h2>
        <div className="gift-list">{this.renderGifts()}</div>
        <Button className="btn-add" onClick={this.addGift}>
          Add gift
        </Button>
      </div>
    );
  }
}

export default App;
