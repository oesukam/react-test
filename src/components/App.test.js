import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("App", () => {
  const app = shallow(<App />);
  test("should render the APP", () => {
    expect(app).toMatchSnapshot();
  });

  test("initialise the `state` with empty array", () => {
    expect(app.state().gifts).toEqual([]);
  });

  describe("when clicking the `add gift` button", () => {
    beforeEach(() => {
      app.find(".btn-add").simulate("click");
    });
    afterEach(() => {
      app.setState({ gifts: [] });
    });
    test("add a new gift to `state`", () => {
      expect(app.state().gifts.length).toBe(1);
    });

    test("add a new gift to the rendered list", () => {
      console.log(app.debug());
      expect(app.find(".gift-list").children().length).toBe(1);
    });
  });
});
