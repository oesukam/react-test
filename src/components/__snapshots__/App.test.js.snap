// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the APP 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Gift Giver
        </h2>,
        <div
          className="gift-list"
        />,
        <ForwardRef
          active={false}
          className="btn-add"
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          Add gift
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Gift Giver",
        },
        "ref": null,
        "rendered": "Gift Giver",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "gift-list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": false,
          "children": "Add gift",
          "className": "btn-add",
          "disabled": false,
          "onClick": [Function],
          "type": "button",
          "variant": "primary",
        },
        "ref": null,
        "rendered": "Add gift",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "active": false,
            "disabled": false,
            "type": "button",
            "variant": "primary",
          },
          "displayName": "Button",
          "render": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Gift Giver
          </h2>,
          <div
            className="gift-list"
          />,
          <ForwardRef
            active={false}
            className="btn-add"
            disabled={false}
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            Add gift
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gift Giver",
          },
          "ref": null,
          "rendered": "Gift Giver",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "gift-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": "Add gift",
            "className": "btn-add",
            "disabled": false,
            "onClick": [Function],
            "type": "button",
            "variant": "primary",
          },
          "ref": null,
          "rendered": "Add gift",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "active": false,
              "disabled": false,
              "type": "button",
              "variant": "primary",
            },
            "displayName": "Button",
            "render": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
